test/test_depsolve.py

```diff
@@ -0,0 +1,223 @@
+import configparser
+import importlib
+import json
+import os
+import socket
+import subprocess as sp
+from tempfile import TemporaryDirectory
+
+import pytest
+
+REPO_PATHS = [
+    "./test/data/testrepos/baseos/",
+    "./test/data/testrepos/custom/",
+]
+
+
+def has_dnf5():
+    return bool(importlib.util.find_spec("libdnf5"))
+
+
+def depsolve(pkgs, repos, repos_dir, cache_dir, command):
+    req = {
+        "command": "depsolve",
+        "arch": "x86_64",
+        "module_platform_id": "platform:el9",
+        "cachedir": cache_dir,
+        "arguments": {
+            "repos_dir": repos_dir,
+            "repos": repos,
+            "transactions": [
+                {"package-specs": pkgs},
+            ]
+        }
+    }
+    p = sp.run([command], input=json.dumps(req).encode(), check=True, capture_output=True)
+    if p.stderr:
+        print(p.stderr.decode())
+
+    return json.loads(p.stdout.decode())
+
+
+def get_rand_port():
+    s = socket.socket()
+    s.bind(("", 0))
+    return s.getsockname()[1]
+
+
+@pytest.fixture(name="repo_servers", scope="module")
+def repo_servers_fixture():
+    procs = []
+    addresses = []
+    for path in REPO_PATHS:
+        port = get_rand_port()  # this is racy, but should be okay
+        p = sp.Popen(["python3", "-m", "http.server", str(port)], cwd=path, stdout=sp.PIPE)
+        procs.append(p)
+        # use last path component as name
+        name = os.path.basename(path.rstrip("/"))
+        addresses.append({"name": name, "address": f"http://localhost:{port}"})
+    yield addresses
+    for p in procs:
+        p.kill()
+
+
+test_cases = [
+    {
+        "packages": ["filesystem"],
+        "results": {
+            "basesystem",
+            "bash",
+            "centos-gpg-keys",
+            "centos-stream-release",
+            "centos-stream-repos",
+            "filesystem",
+            "glibc",
+            "glibc-common",
+            "glibc-minimal-langpack",
+            "libgcc",
+            "ncurses-base",
+            "ncurses-libs",
+            "setup",
+            "tzdata",
+        }
+    },
+    {
+        "packages": ["nothing"],
```

Comment:
maybe a comment here like `# nothing is a package with no dependencies`. Or maybe `nothing`->`pkg-without-deps`?

Reply:
Added a comment.  It was less work than the 3 seconds needed to rebuild the repo metadata :laughing: